src/main/java/at/hannibal2/skyhanni/features/misc/GeorgeDisplay.kt

```diff
@@ -0,0 +1,117 @@
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.InventoryOpenEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.cleanName
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NEUItems.getPriceOrNull
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class GeorgeDisplay {
+
+    private val config get() = SkyHanniMod.feature.misc.pets.george
+
+    private val SEPARATOR: String = ";"
+
+    private val patternGroup = RepoPattern.group("george.tamingcap")
+    private val neededPetPattern by patternGroup.pattern(
+        "needed.pet.loreline",
+        "(?i) +(?<fullThing>(?<tierColorCodes>§.)*(?<tier>(?:un)?common|rare|epic|legendary|mythic|divine|(?:very )?special|ultimate|supreme|admin) (?<pet>[\\S ]+))"
+    )
+
+    private val display = mutableListOf<Renderable>()
+
+    private enum class RarityToTier(
+        val rarityInternal: String,
+        val tier: Int,
+        val rarityDisplay: String,
+    ) {
+        COMMON("COMMON", 0, "§fCommon"),
+        UNCOMMON("UNCOMMON", 1, "§aUncommon"),
+        RARE("RARE", 2, "§9Rare"),
+        EPIC("EPIC", 3, "§5Epic"),
+        LEGENDARY("LEGENDARY", 4, "§6Legendary"),
+        MYTHIC("MYTHIC", 5, "§dMythic"),
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryOpenEvent) {
+        if (!isEnabled()) return
+        if (event.inventoryName != "Offer Pets") return
+        val stack = event.inventoryItems[41] ?: return
+        if (stack.cleanName() != "+1 Taming Level Cap") return
+        display.clear()
+        display.addAll(listBuilding(stack.getLore()))
+    }
+
+    private fun listBuilding(lore: List<String>): MutableList<Renderable> {
+        val updateList: MutableList<Renderable> = mutableListOf<Renderable>(Renderable.string("§d§lTaming 60 Cost: §r§d(${if (config.otherRarities) "cheapest" else "exact"} rarity)"))
+        var totalCost: Double = 0.0
+        for (line in lore) {
+            neededPetPattern.matchMatcher(line) {
+                val origTierString = group("tier") ?: ""
+                val tier = RarityToTier.entries.find { it.rarityInternal == origTierString.uppercase() }?.tier ?: -1
+                val origPetString = group("pet") ?: ""
+                val pet = origPetString.uppercase().replace(" ", "_").removePrefix("FROST_")
+                val petPrices: MutableList<Double> = mutableListOf<Double>()
```

Comment:
<Double>

Reply:
unclear

Reply:
< Double >

this can be removed. some formatting thing thought its a cool idea to simply remove it from this description block lol